### PR TITLE
WIP - Generalized windowed_slo

### PR DESCRIPTION
**Pull request by pauldoomgov**
This heap pares with https://github.com/18F/identity-devops/pull/5634 to provide a more general windowed_slo:

* Drops the metric classes to just two: SingleMetric and CompoundMetric (which just manages a list of SingleMetric objects it sums to derive its own sum)
* Supports `statistic` or `extended_statistic` args to allow for specifying any simple or extended stats from the large list of options
* Lots of high level error handling to make it easy to see what went wrong when the Lambda is sad and allow processing well formed SLIs while just complaining about malformed ones
* The window_days value is moved into the SLI definition to allow for different values
* Adds a couple JSON fixtures to test with
* ...does not yet have test coverage!   (But Paul will gladly add it if you like this thing)

### CODE REVIEW

File: slo_lambda/src/windowed_slo.py [hunk 1]
```diff
@@ -76,164 +56,74 @@ class SingleMetric(MetricBase):
...
+        statistic: Optional[str] = "Sum",
+        extended_statistic: Optional[str] = None,
+        multiplier: float = 1.0,
+    ):
         self.namespace = namespace
         self.metric_name = metric_name
         self.dimensions = dimensions
+        self.statistic = statistic
+        self.extended_statistic = extended_statistic
+        self.multiplier = float(multiplier)
```
Comment (pauldoomgov) [inline slo_lambda/src/windowed_slo.py:74]: @akrito - To allow having metrics that are negated or even scaled relative to others, how about this "multiplier" arg?

File: slo_lambda/src/windowed_slo.py [hunk 2]
```diff
@@ -76,164 +56,74 @@ class SingleMetric(MetricBase):
...
-                **self.stat_args,
-        )['Datapoints']:
+            Namespace=self.namespace,
+            MetricName=self.metric_name,
+            Dimensions=self.dimensions,
+            **self.stat_args,
+        )["Datapoints"]:
             total += self.extract_stat(datapoint)
-        return total
+        return total * self.multiplier
```
Comment (pauldoomgov) [inline slo_lambda/src/windowed_slo.py:100]: @akrito - And here is the only other part needed for the multiplier.

With a multiplier of -1 the output will be subtracted from other values in the composite.

File: slo_lambda/src/windowed_slo.py (hunk 1 above)
Comment (akrito) [reply slo_lambda/src/windowed_slo.py:74]: sgtm

File: slo_lambda/src/windowed_slo.py (hunk 2 above)
Comment (akrito) [reply slo_lambda/src/windowed_slo.py:100]: sweet

**Pull request merged**